### PR TITLE
make json format approvals resilient to differences in pretty json fo…

### PR DESCRIPTION
**Pull request by Goltergaul**
Pr for Issue #84 

This is backwards compatible. The only thing that does not yet work perfectly is showing the diff if there is a real difference in the received json because here it will still show the diff to the raw approved file which might have different whitespaces. Those whitespace differences are then shown in the diff. (e.g. when you have `diff_on_approval_failure` turned on for rspec) But i think thats not such a big problem.

What do you think?

### CODE REVIEW

File: spec/approvals_spec.rb [hunk 1]
```diff
@@ -105,6 +105,11 @@ def filter(data)
     Approvals.verify json, :format => :json, :namer => namer
   end
 
+  it "verifies json where the pretty generation is different but the content is the same" do
```
Comment (markijbema) [inline spec/approvals_spec.rb]: I would also add some tests how diffs show when you changed something, and the pretty generation is different of the approved and received file.

Comment (Goltergaul) [reply spec/approvals_spec.rb]: as far as i saw the diff is generated by rspec by just receiving what is written in `approved.json` and `received.json` Not sure how i could test that.

**Comment by Goltergaul** [comment]
@kytrinyx there is some problem with nokogiri which fails randomly on travis:

> Gem::RuntimeRequirementNotMetError: nokogiri requires Ruby version >= 2.1.0. The
> current ruby version is 2.0.0.
> An error occurred while installing nokogiri (1.8.1), and Bundler
> cannot continue.

**Comment by kytrinyx** [comment]
@Goltergaul Sorry for the slow reply. I've been away for the past 3 weeks, and am catching up on all my notifications today.

I am not sure what the nokogiri failures are about. I wonder if Travis has a chat room or IRC channel to ask them about it.

I'll give this a proper review in the next couple of days!

File: lib/approvals/verifiers/json_verifier.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    class JsonVerifier
+      def initialize(received_path, approved_path)
+        self.received_path = received_path
+        self.approved_path = approved_path
+      end
+
+      def verify
+        return false unless approved == received
+        true
+      end
```
Comment (kytrinyx) [inline lib/approvals/verifiers/json_verifier.rb]: I'm curious why this couldn't be

```diff
       def verify
-        return false unless approved == received
-        true
+       approved == received
       end
```

Comment (Goltergaul) [reply lib/approvals/verifiers/json_verifier.rb]: it definitely can... no idea why i wrote it that way. will change it

**Comment by Goltergaul** [comment]
fails again because of nokogiri. Do we maybe need a version constraint for older rubies somewhere?

**Comment by kytrinyx** [comment]
We may! I'll look into the nokogiri thing. Meanwhile, this looks great, so I'm going to go ahead and merge it, nokokiri be damned.

Thanks for taking the time to do this! I'll cut a release in the next few days so you don't have to work off of a SHA for your project.

**Comment by Goltergaul** [comment]
@kytrinyx cool thank you, could you pls ping me here when you have released the new version.

**Comment by kytrinyx** [comment]
Oh, shoot. This completely fell off my radar. Yes! I will get this done and ping you here. So sorry about that.

**Comment by kytrinyx** [comment]
Alright, v0.0.24 is released (I ended up having to skip v0.0.23 because I bumped the version before discovering that the actual release task would fail to run because of changes to the requirements about gem metadata).

**Comment by Goltergaul** [comment]
awesome thx <3

**Pull request merged**